### PR TITLE
Adds support for Financial Actions API

### PR DESCRIPTION
**Pull request by martinseco**

### CODE REVIEW

**Review by armandojaleo — APPROVED** [review]
👍

File: src/CheckoutSdk/Financial/ActionBreakdown.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+using Checkout.Common;
+using System;
+
+namespace Checkout.Financial
+{
+    public class ActionBreakdown
+    {
+        public string BreakdownType { get; set; }
+
+        public float FxRateApplied { get; set; }
```
Comment (Armando-Ibarra) [inline src/CheckoutSdk/Financial/ActionBreakdown.cs:10]: Please migrate float to double, and also include `?`, all primitives, enums, dates must have the null marker

File: src/CheckoutSdk/Financial/ActionBreakdown.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+namespace Checkout.Financial
+{
+    public class ActionBreakdown
+    {
+        public string BreakdownType { get; set; }
+
+        public float FxRateApplied { get; set; }
+
+        public Currency HoldingCurrency { get; set; }
```
Comment (Armando-Ibarra) [inline src/CheckoutSdk/Financial/ActionBreakdown.cs:12]: Same comment for `?`

File: src/CheckoutSdk/Financial/FinancialClient.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        public FinancialClient(IApiClient apiClient,
+            CheckoutConfiguration configuration) : base(apiClient, configuration, SdkAuthorizationType.SecretKeyOrOAuth)
+        {
+        }
+
+        public Task<FinancialActionsQueryResponse> Query(
+            FinancialActionsQueryFilter query,
+            CancellationToken cancellationToken = default)
+        {
+            return ApiClient.Query<FinancialActionsQueryResponse>(
```
Comment (Armando-Ibarra) [inline src/CheckoutSdk/Financial/FinancialClient.cs:19]: Please add `ValidateParams`

File: src/CheckoutSdk/Financial/FinancialAction.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
+using Checkout.Common;
+using System;
+using System.Collections.Generic;
+
+namespace Checkout.Financial
+{
+    public class FinancialAction
```
Comment (Armando-Ibarra) [inline src/CheckoutSdk/Financial/FinancialAction.cs:7]: Don't forget to add `?` to enums here.

File: test/CheckoutSdkTest/Financial/FinancialIntegrationTest.cs [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+                    action.EntityId.ShouldNotBeNull();
+                    action.CurrencyAccountId.ShouldNotBeNull();
+                    action.ProcessedOn.ShouldNotBeNull();
+                    action.RequestedOn.ShouldNotBeNull();
+                    action.PaymentId.ShouldBe(payment.Id);
+                }
+            }
+        }
+
+        private async Task<PaymentResponse> MakeCardPayment(
```
Comment (Armando-Ibarra) [inline test/CheckoutSdkTest/Financial/FinancialIntegrationTest.cs:53]: I think you can reuse `AbstractPaymentsIntegrationTest`

**Review by Armando-Ibarra — CHANGES_REQUESTED** [review]

**Review by Armando-Ibarra — APPROVED** [review]
Good job!

**Pull request merged**